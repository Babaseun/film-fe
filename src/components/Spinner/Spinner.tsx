import React from 'react';
import './Spinner.scss';

function Spinner() {
  return (
    <div className="spinner">
      <p data-testid="spinner">
        Loading.... <i className="fas fa-spinner fa-spin fa-10x" />
      </p>
    </div>
  );
}

export default Spinner;
