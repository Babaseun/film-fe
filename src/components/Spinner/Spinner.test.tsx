import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Spinner from './Spinner';

describe('<Spinner/>.', () => {
  it('renders spinner component', () => {
    render(<Spinner />);
    const spinner = screen.getByTestId('spinner');
    expect(spinner).toHaveTextContent('Loading');
  });
  it('matches snapshot', () => {
    const tree = renderer
      .create(
        <div className="spinner">
          <p data-testid="spinner">
            Loading.... <i className="fas fa-spinner fa-spin fa-10x" />
          </p>
        </div>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
