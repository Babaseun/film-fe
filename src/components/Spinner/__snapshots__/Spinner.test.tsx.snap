// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner/>. matches snapshot 1`] = `
<div
  className="spinner"
>
  <p
    data-testid="spinner"
  >
    Loading.... 
    <i
      className="fas fa-spinner fa-spin fa-10x"
    />
  </p>
</div>
`;
